### PR TITLE
Add contains method to LocalCheckpointTracker

### PR DESCRIPTION
**Pull request by dnhatn**
This change adds "contains" method to LocalCheckpointTracker.
One of the use cases is to check if a given operation has been processed 
in an engine or not by looking up its seq_no in LocalCheckpointTracker.

Relates #33656

### CODE REVIEW

**Review by s1monw — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/index/seqno/LocalCheckpointTracker.java [hunk 1]
```diff
@@ -158,6 +158,21 @@ public synchronized void waitForOpsToComplete(final long seqNo) throws Interrupt
         }
     }
 
+    /**
+     * Checks if the given sequence number was marked as completed in this tracker.
+     */
+    public synchronized boolean contains(long seqNo) {
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/index/seqno/LocalCheckpointTracker.java]: I don't think we need to be fully synchronized here (and so can save ourselves from locking for the quick returns). I think that we only need to lock on this on access to `processedSeqNo`. I see that we have asserts that we have a lock on this in `getBitSetKey` and `seqNoToBitSetOffset` but I think that neither of those are necessary.

**Review by jasontedor — COMMENTED** [review]
I left a comment.

File: server/src/main/java/org/elasticsearch/index/seqno/LocalCheckpointTracker.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/index/seqno/LocalCheckpointTracker.java]: Good suggestion. I pushed https://github.com/elastic/elasticsearch/pull/33871/commits/d22f136a757e47fd427a5a897c9603313cfc5d5d.

**Review by jasontedor — APPROVED** [review]
Looks good @dnhatn! Thanks.

**Comment by dnhatn** [comment]
Thanks @s1monw and @jasontedor.

**Pull request merged**